shell/common/shell.cc

```diff
@@ -416,6 +371,7 @@ fml::WeakPtr<Rasterizer> Shell::GetRasterizer() {
 }
 
 fml::WeakPtr<Engine> Shell::GetEngine() {
+  ui_latch_.Wait();
```

Comment:
I partly wonder if a spinlock would be better here so that later callers won't face a potential context switch when aquiring the mutex here - but that may not be beneficial during startup. It'd be interesting to come up with a benchmark for that and see if it's worth pursuing.  Alternatively, maybe we could have some kind of `atomic_bool` or something that we set after signaling this latch so that we don't have to wait on it every time afterward.

Reply:
good idea.